perun-base/src/main/java/cz/metacentrum/perun/oidc/UserInfoEndpointCall.java

```diff
@@ -0,0 +1,110 @@
+package cz.metacentrum.perun.oidc;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import cz.metacentrum.perun.core.api.BeansUtils;
+import cz.metacentrum.perun.core.api.exceptions.InternalErrorException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.*;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.testcontainers.shaded.org.apache.commons.lang3.StringUtils.isEmpty;
+import static org.testcontainers.shaded.org.apache.commons.lang3.StringUtils.isNotEmpty;
+
+public class UserInfoEndpointCall {
+
+	private final static Logger log = LoggerFactory.getLogger(UserInfoEndpointCall.class);
+
+	public UserInfoEndpointResponse getUserInfoEndpointData(String accessToken, String issuer, Map<String, String> additionalInformation) throws ExpiredTokenException {
```

Comment:
This method seems to be quite long. I would recommend to extract some of its parts to separate methods for better readability. For example, you might put retrieving of each part to separate method. What do you think?